kids/kid0003.md

```diff
@@ -396,29 +403,82 @@ seal is event seal of latest establishment event of validator
+
 ```javascript
 [
   "sith",  
   "keys",  
 ]
 
 ```
+The next signing config has two primary elements, sith and keys. The combination starts with the sith as the first element. The keys element is then combined. But because the keys element is a list, each key from the list must be combined in order of appearance with the combination of the prior elements. The "sith" element may also be a list or list of lists when using a fractionally weighted threshold.
+
+The combination method is to first take a digest of each element and then combine the elements with an XOR. Finally the combined result is converted to a fully qualified Base64 representation with the derivation code for the type of digest used. All elements must use the same digest.
+
+The sith element is converted to a string before digesting. For an integer sith, the string is a hex representation of the integer with no leading zeros.
+
+Each of the keys is first converted to fully qualified Base64 format which is the normal format for the keys in key events. The the digest is on this fully qualified Base64 format.  The digests, sith plus each of the keys, are combined in raw binary mode using a bitwise XOR. The combined digests are then converted to fully qualified Base64 and this is the format used in the key event for the value of the nxt element.
+
+#### Example computation
+The following example shows values.
+
+``` python
+sith = 2
+keys = ['BrHLayDN-mXKv62DAjFLX1_Y5yEUe0vA9YPe_ihiKYHE',
+        'BujP_71bmWFVcvFmkE9uS8BTZ54GIstZ20nj_UloF8Rk',
+        'B8T4xkb8En6o0Uo5ZImco1_08gT5zcYnXzizUPVNzicw']
+            
+# convert sith to hex string no leading zeros
+
+sith = '2'
+
+# sith digest
+# fully qualified
+sithdig = 'EgT6bcpFB5_OFr6Ci0N8-bDeJ5Cf_5K7vVmpWW8jy_j0'
+
+# raw binary unqualified in string escaped form
+sithdig == b"\x81>\x9br\x91A\xe7\xf3\x85\xaf\xa0\xa2\xd0\xdf>l7\x89\xe4'\xff\xe4\xae\xefVjV[\xc8\xf2\xfe="
+
+# convert keys to digests
+# fully qualified Base64 digests
+keydigs == ['EmB26yMzroICh-opKNdkYyP000kwevU18WQI95JaJDjY',
+         'EO4CXp8gs0yJg1fFhJLs5hH6neqJwhFEY7vrJEdPe87I',
+         'ELWWZEyBpjrfM1UU0n31KIyIXllrCoLEOI5UHD9x7WxI']
+         
+# raw binary (unqualifed) digests in string escaped form
+keydigs =[b'\x98\x1d\xba\xc8\xcc\xeb\xa0\x80\xa1\xfa\x8aJ5\xd9\x18\xc8\xfd4\xd2L\x1e\xbdM|Y\x02=\xe4\x96\x89\x0e6',
+                       b';\x80\x97\xa7\xc8,\xd3"`\xd5\xf1a$\xbb9\x84~\xa7z\xa2p\x84Q\x18\xee\xfa\xc9\x11\xd3\xde\xf3\xb2',
+                       b'-e\x99\x13 i\x8e\xb7\xcc\xd5E4\x9f}J#"\x17\x96Z\xc2\xa0\xb1\x0e#\x95\x07\x0f\xdc{[\x12']
+
+# now combine with XOR the raw versions of all four digests, sith plus three keys. In python must first convert to integers in order to use XOR = ^
+
+raw = sithdig ^ keydig[0] ^ keydig[1] ^ keydig[2]
+
+# now convert raw to fully qualified with derivation code for the digest algorithm
+# in this case Blake3_256
+
+nxt = 'ED8YvDrXvGuaIVZ69XsBVA5YN2pNTfQOFwgeloVHeWKs'
```

Comment:
All keys must be digested using the same algorithm and that algorithm is indicated by the derivation code, have I got that right?

Reply:
Yes thats right.